docs/guides/tutorials/nextjs.rst

```diff
@@ -0,0 +1,270 @@
+
+   .. code-block:: bash
+
+     $ npm install edgedb
+     # or yarn add edgedb
+     # or pnpm add edgedb
+
+3. Install the query builder generator (recommended for TypeScript projects):
+
+   .. code-block:: bash
+
+     $ npm install --save-dev @edgedb/generate
+     # or yarn add --dev @edgedb/generate
+     # or pnpm add --dev @edgedb/generate
+
+Updating the schema
+-------------------
+
+Start by adding new types to your EdgeDB schema:
+
+.. code-block:: sdl
+  :caption: dbschema/default.esdl
+
+  module default {
+    type BlogPost {
+      required title: str;
+      required content: str {
+        default := ""
+      }
+    }
+  }
+
+Then create and apply your first migration:
+
+.. code-block:: bash
+
+   $ npx edgedb migration create
+   $ npx edgedb migrate
+
+You can now run queries against your new schema. Run the following command to open the EdgeDB REPL:
+
+.. code-block:: bash
+
+  $ npx edgedb
+
+You can then execute the following ``insert`` statements:
+
+.. code-block:: edgeql-repl
+
+  edgedb> insert BlogPost {
+  .......   title := "This one weird trick makes using databases fun",
+  .......   content := "Use EdgeDB"
+  ....... };
+  {default::BlogPost {id: 7f301d02-c780-11ec-8a1a-a34776e884a0}}
+  edgedb> insert BlogPost {
+  .......   title := "How to build a blog with EdgeDB and Next.js",
+  .......   content := "Let's start by scaffolding our app..."
+  ....... };
+  {default::BlogPost {id: 88c800e6-c780-11ec-8a1a-b3a3020189dd}}
+
+
+Alternatively, you can use the EdgeDB UI to manage your schema and data. Open the EdgeDB UI by running:
+
+.. code-block:: bash
+
+  $ npx edgedb ui
+
+
+Generating the query builder​ 
+-----------------------------
+
+This step is optional but recommended for TypeScript projects. The query builder generates TypeScript types for your EdgeQL queries:
+
+.. code-block:: bash
+
+  $ npx @edgedb/generate edgeql-js
+
+The command introspects your schema and generates a query builder in the ``dbschema/edgeql-js`` directory.
+
+Using EdgeDB in Next.js
+-----------------------
+
+EdgeDB with React Server Components (App Router)
+================================================
+
+Create a server component that fetches data directly from EdgeDB:
+
+.. code-block:: tsx
+  :caption: app/page.tsx
+
+  import { createClient } from 'edgedb';
+  import e from '@/dbschema/edgeql-js';
+
+  const client = createClient();
+
+  export default async function Posts() {
+    const posts = await e.select(e.BlogPost, () => ({
+      id: true,
+      title: true,
+      content: true,
+    })).run(client);
+
+    return (
+      <div>
+        {posts.map(post => (
+          <article key={post.id}>
+            <h2>{post.title}</h2>
+            <p>{post.content}</p>
+          </article>
+        ))}
+      </div>
+    );
+  }
+
+With API Routes
+===============
+
+Create an API route and fetch data from the client side:
+
+.. code-block:: tsx
+  :caption: pages/api/posts.ts
```

Comment:
We should update this to be an app-router Route Handler, right?

Reply:
I was thinking to have two options there - with app router and with pages router. I'll make it clear in the title.

Reply:
Do we want to support the pages router still? There is nothing particularly special about this code, and if we show an app router Route Handler, it should be pretty easy for someone two write the equivalent pages router API route, I would think.